tests/integrations/raftstore/test_lease_read.rs

```diff
@@ -312,3 +312,85 @@ fn test_node_callback_when_destroyed() {
         resp
     );
 }
+
+#[test]
+fn test_read_index_when_transfer_leader() {
+    let mut cluster = new_node_cluster(0, 3);
+
+    // Increase the election tick to make this test case running reliably.
+    configure_for_lease_read(&mut cluster, Some(50), Some(10_000));
+    let max_lease = Duration::from_secs(2);
+    cluster.cfg.raft_store.raft_store_max_leader_lease = ReadableDuration(max_lease);
+
+    cluster.pd_client.disable_default_operator();
+    let r1 = cluster.run_conf_change();
+    cluster.must_put(b"k1", b"v1");
+    cluster.pd_client.must_add_peer(r1, new_peer(2, 2));
+    cluster.pd_client.must_add_peer(r1, new_peer(3, 3));
+    must_get_equal(&cluster.get_engine(3), b"k1", b"v1");
+
+    // Put and test again to ensure that peer 3 get the latest writes by message append
+    // instead of snapshot, so that transfer leader to peer 3 can 100% success.
+    cluster.must_put(b"k1", b"v2");
+    sleep_ms(100);
```

Comment:
Why sleep?

Reply:
Otherwise `must_get_equal` under the line fails frequently.

Reply:
Why? The function sleeps at most 5s already.